services/web/server/tests/unit/with_dbs/10/meta/test_meta_iterations.py

```diff
@@ -0,0 +1,232 @@
+        project_iter0 = body["data"]
+
+        outputs = {}
+        for nid, node in project_iter0["workbench"].items():
+            if out := node.get("outputs"):
+                outputs[nid] = out
+
+        assert len(outputs) == 1
+        assert outputs["fc9208d9-1a0a-430c-9951-9feaf1de3368"]["out_1"] == i
+
+    # ----------------------------------------------
+
+    # GET project and MODIFY iterator values----------------------------------------------
+    #  - Change iterations from 0:4 -> HEAD+1
+    resp = await client.get(f"/v0/projects/{project_uuid}")
+    assert resp.status == HTTPStatus.OK, await resp.text()
+    body = await resp.json()
+
+    # TODO: updating a project fields can be daunting because
+    # it combines nested field attributes with dicts and from the
+    # json you cannot distinguish easily what-is-what automatically
+    # Dict keys are usually some sort of identifier, typically a UUID or
+    # and index but nothing prevents a dict from user other type of key types
+    #
+    project = Project.parse_obj(body["data"])
+    new_project = project.copy(
+        update={
+            # FIXME: HACK to overcome export from None -> string
+            # SOLUTION 1: thumbnail should not be required (check with team!)
+            # SOLUTION 2: make thumbnail nullable
+            "thumbnail": faker.image_url(),
+        }
+    )
+    assert new_project.workbench is not None
+    assert new_project.workbench
+    node = new_project.workbench["fc9208d9-1a0a-430c-9951-9feaf1de3368"]
+    assert node.inputs
+    node.inputs["linspace_stop"] = 4
+
+    resp = await client.put(
+        f"/v0/projects/{project_uuid}",
+        data=json_dumps(new_project.dict(**REQUEST_MODEL_POLICY)),
+    )
+    assert resp.status == HTTPStatus.OK, await resp.text()
+
+    # RUN again them ---------------------------------------------------------------------------
+    resp = await client.post(
+        f"/v0/computation/pipeline/{project_uuid}:start",
+        json=RUN_PROJECT.request_payload,
+    )
+    data, _ = await assert_status(resp, web.HTTPCreated)
+    assert project_uuid == data["pipeline_id"]
+    ref_ids = data["ref_ids"]
+    assert len(ref_ids) == 4
+
+    # GET iterations -----------------------------------------------------------------
+    # check iters 1, 2 and 3 share working copies
+    #
+    resp = await client.get(f"/v0/repos/projects/{project_uuid}/checkpoints/HEAD")
+    body = await resp.json()
+    head_ref_id = body["data"]["id"]
+
+    assert head_ref_id == 5
+
+    resp = await client.get(
+        f"/v0/projects/{project_uuid}/checkpoint/{head_ref_id}/iterations?offset=0"
+    )
+    body = await resp.json()
+    second_iterlist = Page[ProjectIterationAsItem].parse_obj(body).data
+
+    assert len(second_iterlist) == 4
+    assert len(set(it.workcopy_project_id for it in second_iterlist)) == len(
+        second_iterlist
+    ), "unique"
+
+    # TODO: cached iterations will be implemented in next PR
```

Comment:
still relevant?

Reply:
This is coming in next PR. I would like to keep it as a reminder

Reply:
added a link to all those